app/controllers/benchmark_runs_controller.rb

```diff
@@ -31,6 +31,10 @@ def create
     benchmark_run.result = params[:benchmark_run][:result]
     benchmark_run.save!
 
+    if params[:commit_hash]
+      BenchmarkRegressionJob.perform_later(benchmark_run, initiator, benchmark_type, benchmark_result_type, benchmark_type_params[:category])
```

Comment:
I have reduced to only 3 parameters `benchmark_run, benchmark_type, benchmark_result_type`. Doing something as shown in the link will be difficult for us as all this parameters are obtained as a result of many different methods.